grafana-plugin/src/components/WithConfirm/WithConfirm.tsx

```diff
@@ -1,18 +1,22 @@
 import React, { ReactElement, useCallback, useState } from 'react';
 
-import { ConfirmModal } from '@grafana/ui';
+import { ConfirmModal, ConfirmModalProps } from '@grafana/ui';
 
-interface WithConfirmProps {
+type WithConfirmProps = Partial<ConfirmModalProps> & {
   children: ReactElement;
   title?: string;
-  body?: React.ReactNode;
   confirmText?: string;
   disabled?: boolean;
-}
-
-const WithConfirm = (props: WithConfirmProps) => {
-  const { children, title = 'Are you sure to delete?', body, confirmText = 'Delete', disabled } = props;
+};
 
+const WithConfirm: React.FC<WithConfirmProps> = ({
+  children,
+  title = 'Are you sure to delete?',
+  confirmText = 'Delete',
+  body,
+  disabled,
+  ...props
```

Comment:
why do you explicitly extract `title` `body` etc...and hide `description` and `confirmationText ` behind `...props`, probably adding `description` and `confirmationText` to destructuring is better to avoid confusing

Reply:
`children` and `disabled` need to be extracted because they're not passed down to `ConfirmModal`. `title` and `confirmText` as well, since they have default values.

As for the rest, those props simply get passed along to `ConfirmModal`.